sdk/identity/Azure.Identity/src/MsalPublicClient.cs

```diff
@@ -74,25 +74,30 @@ public virtual async ValueTask<AuthenticationResult> AcquireTokenSilentAsync(str
         public virtual async ValueTask<AuthenticationResult> AcquireTokenInteractiveAsync(string[] scopes, string claims, Prompt prompt, bool async, CancellationToken cancellationToken)
         {
             IPublicClientApplication client = await GetClientAsync(async, cancellationToken).ConfigureAwait(false);
-            if (async)
+
+#pragma warning disable AZC0109 // Misuse of 'async' parameter.
+            if (!async && !IdentityCompatSwitches.DisableInteractiveBrowserThreadpoolExecution)
```

Comment:
Is there a reasonable way to test this switch behavior?

Reply:
I was planning on adding a test for the `IdentityCompatSwitches` directly to verify we're reading the value properly. I could also add a test hook of some sort to allow us to verify we honor it here. If you have thoughts on a reasonable approach I'd welcome ideas.

Reply:
In the past I wrote an `IDisposable` that saved current and set a new environment variable, then on `Dispose` reset it. Unless the process crashes, it has generally worked well.